sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs

```diff
@@ -537,11 +538,17 @@ internal static BlobStorageResourceOptions GetBlobResourceOptions(
             return new()
             {
                 Metadata = checkpointDetails.Metadata,
+                _isMetadataSet = checkpointDetails.IsMetadataSet,
```

Comment:
How would you feel about maybe providing an internal constructor to `BlobStorageResourceOptions` that takes all properties and the bool for each property and sets the private members. It feels weird to be setting the "should be private" members of the properties directly. This could be easy to mess up in the future, like someone sets metadata but forgets to set the _isMetadataSet.